File: solutions/java/word-count/1/src/main/java/WordCount.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Objects;
import java.util.function.Function;
import java.util.stream.Collectors;

class WordCount {
    public Map<String, Integer> phrase(String text) {
        return Arrays.stream(text.toLowerCase().replaceAll("[^a-z0-9']+", " ")
                        .split("\\s+"))
                .filter(s -> !s.isEmpty())
                .collect(Collectors.groupingBy(
                        Function.identity(),
                        LinkedHashMap::new,
                        Collectors.summingInt(val -> 1)
                ));
    }
}

File: solutions/typescript/word-count/1/word-count.ts
export function count(text: string): Map<string, number> {
  const clean = text.replace(/[^\w']+|'(?![A-Za-z])|(?<![A-Za-z])'/g, " ").toLowerCase()
  const res = clean.split(" ").filter(x => x != "")
      .reduce((acc, word) => {
        if (acc.has(word)) {
          const val: number = acc.get(word) ?? 0
          acc.set(word, val + 1)
        } else {
          acc.set(word, 1)
        }
        return acc
      }, new Map())

  console.log(res)
  return res
}

